src/Middleware/RateLimiting/test/RateLimitingMiddlewareTests.cs

```diff
@@ -0,0 +1,89 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Moq;
+
+namespace Microsoft.AspNetCore.RateLimiting;
+public class RateLimitingMiddlewareTests : LoggedTest
+{
+    [Fact]
+    public void Ctor_ThrowsExceptionsWhenNullArgs()
+    {
+        var options = CreateOptionsAccessor();
+        options.Value.Limiter = new TestPartitionedRateLimiter<HttpContext>();
+
+        Assert.Throws<ArgumentNullException>(() => new RateLimitingMiddleware(
```

Comment:
These would never be true though, would it? The ctor is created via DI, and isn't available to user code. So you don't really need to test this.